src/ngraph/runtime/cpu/mkldnn_emitter.cpp

```diff
@@ -138,6 +138,1113 @@ std::vector<size_t>& MKLDNNEmitter::get_primitive_deps(size_t index)
     return m_primitive_deps.at(index);
 }
 
+mkldnn::memory::desc
+    MKLDNNEmitter::build_blocked_memory_descriptor(const mkldnn::memory::dims& dim,
+                                                   const mkldnn::memory::dims& strides,
+                                                   mkldnn::memory::data_type dtype) const
+{
+#if defined(USE_MKLDNN_V1)
```

Comment:
I really don't like the scattered `#if` all-over the place. I would suggest creating two files `mkldnn_emitter_0_x.cpp.in` and `mkldnn_emitter_1_0.cpp.in` and #include them here based on the compile-time flag. 
What do you think ?

Reply:
Or change the CMakeList to choose the right file.

Reply:
I can remove the #if inside of any function and write two functions instead. That way, only one #if is needed. There are a lot of other files need to be changes. Couple of #if would be fine in one file?